File: problems/1. Two Sum/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut seen = HashMap::new();
        for (index, num) in nums.iter().enumerate() {
            if let Some(x) = seen.get(num) {
                return vec![index as i32, *x];
            } else {
                seen.insert(target - *num, index as i32);
            }
        }
        vec![]
    }
}

File: problems/1. Two Sum/solution.py
from ast import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_table={}
        for i in range(len(nums)):
            if (hash_table.get(nums[i])!=None):
                return [i,hash_table[nums[i]]]
            else:
                hash_table[target-nums[i]]=i
    
    # time complexity  : O(n)
    # space complexity : O(n)